libcaf_net/caf/net/socket.hpp

```diff
@@ -24,20 +24,55 @@
 
 #include "caf/config.hpp"
 #include "caf/fwd.hpp"
-#include "caf/net/abstract_socket.hpp"
 #include "caf/net/socket_id.hpp"
 
 namespace caf {
 namespace net {
 
 /// An internal endpoint for sending or receiving data. Can be either a
 /// ::network_socket, ::pipe_socket, ::stream_socket, or ::datagram_socket.
-struct socket : abstract_socket<socket> {
-  using super = abstract_socket<socket>;
+struct socket {
+  socket_id id;
 
-  using super::super;
+  constexpr socket() : id(invalid_socket_id) {
+    // nop
+  }
+
+  constexpr explicit socket(socket_id id) : id(id) {
+    // nop
+  }
+
+  constexpr socket(const socket& other) : id(other.id) {
```

Comment:
Wouldn't `= default;` be enough for this trivial copy-constructor?

Reply:
Absolutely. 🙂